apps/io-services-cms-webapp/src/utils/subscription.ts

```diff
@@ -0,0 +1,74 @@
+import { ApiManagementClient } from "@azure/arm-apimanagement";
+import {
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorTooManyRequests,
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorInternal,
+} from "@pagopa/ts-commons/lib/responses";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as TE from "fp-ts/lib/TaskEither";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { IConfig } from "../config";
+import { getSubscription } from "../lib/clients/apim-client";
+import { MANAGE_APIKEY_PREFIX } from "./api-keys";
+
+type ErrorResponses =
+  | IResponseErrorNotFound
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+  | IResponseErrorTooManyRequests;
+
+const isManageKey = (ownerSubscriptionId: NonEmptyString) =>
+  ownerSubscriptionId.startsWith(MANAGE_APIKEY_PREFIX);
+
+/**
+ * Using the **API Manage key** as 'Ocp-Apim-Subscription-Key', the Subscription relating to this key will have a name starting with "MANAGE-"
+ * and accordingly no longer equal to the serviceId.
+ *
+ * Therefore, since it is no longer possible to verify the equality *subscriptionId == serviceId*,
+ * it is necessary to verify that the owner of the subscription of the API Key is the same owner of the Subscription to which the ServiceId belongs
+ *
+ * @param config
+ * @param apimClient
+ * @param serviceId
+ * @param ownerSubscriptionId subscriptionId related to 'Ocp-Apim-Subscription-Key'
+ * @param userId APIM userId
+ * @returns
+ */
+export const serviceOwnerCheckManageTask = (
+  config: IConfig,
+  apimClient: ApiManagementClient,
+  serviceId: NonEmptyString,
+  ownerSubscriptionId: NonEmptyString,
+  userId: NonEmptyString
+): TaskEither<ErrorResponses, NonEmptyString> =>
+  pipe(
+    ownerSubscriptionId,
+    TE.fromPredicate(
+      (ownSubId: NonEmptyString) => isManageKey(ownSubId),
+      () => ResponseErrorForbiddenNotAuthorized
+    ),
+    TE.chainW(() =>
+      pipe(
+        getSubscription(
+          apimClient,
+          config.AZURE_APIM_RESOURCE_GROUP,
+          config.AZURE_APIM,
+          serviceId
+        ),
+        TE.mapLeft(() =>
+          ResponseErrorInternal(
+            `An error has occurred while retrieving service '${serviceId}'`
+          )
+        )
+      )
+    ),
+    TE.chainW((serviceSubscription) =>
+      serviceSubscription.ownerId === userId
+        ? TE.of(serviceId)
```

Comment:
**suggestion(non-blocking):** IMHO this way it's more readable 😉 
```suggestion
        ? TE.right(serviceId)
```
Otherwise you can use a TE.fromPredicate instead of ternary operator

Reply:
suggestion included in commit [suggestion code cleanUp](https://github.com/pagopa/io-services-cms/pull/139/commits/28f33caeae8a6690bf03b2c2483f459a56dafecd)